backend/data.go

```diff
@@ -159,8 +159,20 @@ type DataResponse struct {
 
 	// Status codes map to HTTP status values
 	Status Status
+
+	// Source is the the source of the error
+	ErrorSource ErrorSource
 }
 
+// ErrorSource type defines the source of the error
+type ErrorSource string
+
+const (
+	ErrorSourceDatabase ErrorSource = "database"
+	ErrorSourcePlugin   ErrorSource = "plugin"
+	ErrorSourceExternal ErrorSource = "external"
```

Comment:
How do we differentiate between external and database?

Reply:
These are based on the [prometheus sources](https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/instrumentation/instrumentation.go#L26).  @bergquist do you know offhand how prometheus differentiates database vs external? I haven't really been in that code.

Reply:
Looked a [bit more](https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/instrumentation/instrumentation.go#L68).  Looks like "external" is really just auth related.

Reply:
It would simplify if we used similar sources as Grafana itself if possible. Can we combine database and external into downstream perhaps? In general and in regards to SLO's plugin authors would only be interested in source = plugin, right?

Reply:
I remember discussing this with Prometheus devs and seemed like they had additional requirements besides SLO's. If a plugin need to track more granular details I would suggest each plugin implementing custom metrics and not mix it up with the general plugin request metrics. WDYT?

Reply:
updated the pr